src/mtype.d

```diff
@@ -7963,13 +7948,14 @@ public:
                 return oe;
             return new DotExp(e.loc, e, oe);
         }
+
         Declaration d = s.isDeclaration();
-        debug
+        if (!d)
         {
-            if (!d)
-                printf("d = %s '%s'\n", s.kind(), s.toChars());
+            e.error("%s.%s is not a declaration", e.toChars(), ident.toChars());
```

Comment:
One idea to emphasize it's not expected to reach the error, is use special error message "Internal Compiler Error: ...". How about that?